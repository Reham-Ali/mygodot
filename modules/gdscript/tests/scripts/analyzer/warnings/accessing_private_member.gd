class A:
	@warning_ignore("unused_private_class_variable")
	var __a
	static var __b

	func __c():
		pass
	static func __d():
		pass

class B:
	var a = A.new()
	var b = a.__a
	var c = A.__b

	func foo():
		@warning_ignore("standalone_expression")
		a.__a
		@warning_ignore("standalone_expression")
		A.__b

		a.__c()
		A.__d()

	var d = a.__c()
	var e = A.__d()

class C extends A:
	var a = A.new()
	var b = a.__a
	var c = A.__b

	func foo():
		@warning_ignore("standalone_expression")
		a.__a
		@warning_ignore("standalone_expression")
		A.__b

		a.__c()
		A.__d()

	var d = a.__c()
	var e = A.__d()

func test():
	pass
