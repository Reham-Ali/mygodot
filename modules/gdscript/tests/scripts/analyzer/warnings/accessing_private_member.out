GDTEST_OK
>> WARNING
>> Line: 13
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an class that is not authorized to access it.
>> WARNING
>> Line: 14
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an class that is not authorized to access it.
>> WARNING
>> Line: 18
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an class that is not authorized to access it.
>> WARNING
>> Line: 20
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an class that is not authorized to access it.
>> WARNING
>> Line: 22
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 23
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an class that is not authorized to call it.
>> WARNING
>> Line: 25
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 26
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an class that is not authorized to call it.
>> WARNING
>> Line: 30
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an class that is not authorized to access it.
>> WARNING
>> Line: 31
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an class that is not authorized to access it.
>> WARNING
>> Line: 35
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an class that is not authorized to access it.
>> WARNING
>> Line: 37
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an class that is not authorized to access it.
>> WARNING
>> Line: 39
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 40
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an class that is not authorized to call it.
>> WARNING
>> Line: 42
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 43
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an class that is not authorized to call it.
